### ISSUE TITLE
[SQL] Generated column getters are annotated with @NonNull for generated columns

### ISSUE THREAD

**Issue by EotT123** [manifold-systems/manifold#562]
Generated column getters are annotated with `@NonNull` for generated columns (e.g. id columns):

Example:
```
@ActualName("id")
default @NotNull Integer getId() {
  return (Integer)this.getBindings().get("id");
}
```

However, when the object is not yet persisted, the value **is** still null.
When verifying if the value is not null, IntelliJ warns me that the value cannot be null: `Condition '<object>.id != null' is always 'true'`

The annotation is added [here](https://github.com/manifold-systems/manifold/blob/ac152bf67803442cf7d3e4e795155c35c1bd9a6b/manifold-deps-parent/manifold-sql/src/main/java/manifold/sql/schema/type/SchemaParentType.java#L923). Shouldn't there be an extra check on generated column values?
`getter.addAnnotation( col.isNullable() && !col.isGenerated() ? Nullable.class.getSimpleName() : NotNull.class.getSimpleName() );`

**Comment by rsmckinney** [comment]
Fix available with release 2024.1.11

**Comment by EotT123** [comment]
Thanks!
